org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -169,6 +175,22 @@ import static org.eclipse.xtext.diagnostics.Severity.*
 					commands = this.commandRegistry.getCommands()
 				]
 			}
+
+			val semanticHighlighting = params?.capabilities?.textDocument?.semanticHighlightingCapabilities?.semanticHighlighting ?: false;
+			if (semanticHighlighting) {
+				val scopes = allLanguages
+					.map[get(ISemanticHighlightingCalculator) -> get(ISemanticHighlightingIdMapper)]
+					.filter[key !== null && value !== null]
+					.map[pair | pair.key.highlightingStyles.map[pair.value.map(it)]]
+					.flatten
+					.filter[!nullOrEmpty]
+					.toList;
+				semanticHighlightingManager.initialize(scopes);
```

Comment:
any reasoning behind the naming scheme. here its called manager for commands its called registry.

Reply:
No, in fact, I like the registry better. I do not mind renaming it.

Reply:
@svenefftinge what do you think?